app/src/main/java/com/larryhsiao/nyx/tags/NewTag.java

```diff
@@ -0,0 +1,39 @@
+package com.larryhsiao.nyx.tags;
+
+import com.silverhetch.clotho.Source;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+import static java.sql.Statement.RETURN_GENERATED_KEYS;
+
+/**
+ * Source to build a Tag that just created.
+ */
+public class NewTag implements Source<Tag> {
+    private final Source<Connection> connSource;
+    private final String title;
+
+    public NewTag(Source<Connection> connSource, String title) {
+        this.connSource = connSource;
+        this.title = title;
+    }
+
+    @Override
+    public Tag value() {
+        try (PreparedStatement stmt = connSource.value().prepareStatement(
+            // language=H2
+            "INSERT INTO TAGS (TITLE)VALUES ( ? );"
+            , RETURN_GENERATED_KEYS
+        )) {
+            stmt.setString(1, title);
+            stmt.executeUpdate();
+            ResultSet res = stmt.getGeneratedKeys();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Always check the return of one of the navigation method (next,previous,first,last) of a ResultSet.](https://app.codacy.com/manual/LarryHsiao/Nyx/pullRequest?prid=5029215)